### PR TITLE
Add a linter

### PR DESCRIPTION
**Pull request by dgageot**
I wanted to start introducing a meta-linter. It won't run on Travis yet, nor does it make sure that the required tool is installed.

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 	lifecycleVersion := descriptor.Info.Version
 	if lifecycleVersion == nil {
 		c.logger.Warnf("lifecycle version unknown, assuming %s", style.Symbol(builder.AssumedLifecycleVersion))
-		lifecycleVersion = builder.VersionMustParse(builder.AssumedLifecycleVersion)
+		// lifecycleVersion = builder.VersionMustParse(builder.AssumedLifecycleVersion)
```
Comment (jromero) [inline build.go]: This doesn't look right.

File: internal/logging/logging.go [hunk 2]
```diff
@@ -112,12 +112,10 @@ func (lw *logWithWriters) WantQuiet(f bool) {
 }
 
 // NewLogWithWriters creates a logger to be used with pack CLI.
-func NewLogWithWriters() *logWithWriters {
+func NewLogWithWriters() *logWithWriters { //nolint:golint
```
Comment (jromero) [inline internal/logging/logging.go]: Can we ignore specific errors instead of all rules from a certain linter?

File: Makefile [hunk 3]
```diff
@@ -30,6 +30,10 @@ format: install-goimports
 	@echo "> Formating code..."
 	@goimports -l -w -local ${PACKAGE_BASE} ${SRC}
 
+lint:
+	@echo "> Linting code..."
+	@golangci-lint run -c golangci.yaml
```
Comment (jromero) [inline Makefile]: We should be able to [automatically install](https://github.com/golangci/golangci-lint#windows-macos-and-linux) this similar to `goimports` and `mockgen`. Is there a reason we wouldn't want to do that?

Comment (jromero) [reply Makefile]: Re-read your description and noticed that you called out that this aspect isn't done yet. 👍

File: build.go (hunk 1 above)
Comment (dgageot) [reply build.go]: yeah. Exactly the kind of bugs that a linter catches :-)

File: internal/logging/logging.go (hunk 2 above)
Comment (dgageot) [reply internal/logging/logging.go]: Yes, I think. I'll try it. The doc says something like that is valid: `//nolint:golint,unused`

File: build.go (hunk 1 above)
Comment (jromero) [reply build.go]: Sorry, I probably should have been more specific. I was calling out the fact that this line is commented out. What lint warning or error was the linter providing?

Comment (dgageot) [reply build.go]: It says, lifecycleVersion is not used:
```
ineffassign: Detects when assignments to existing variables are not used
```

Comment (jromero) [reply build.go]: Ah! That makes sense. Probably remnants of prior logic. The message still makes sense but might even need to move to a more appropriate location of the code base.

So this should probably now look like this for the time being:

```go
if descriptor.Info.Version == nil {
	c.logger.Warnf("lifecycle version unknown, assuming %s", style.Symbol(builder.AssumedLifecycleVersion))
} else {
	c.logger.Debugf("Executing lifecycle version %s", style.Symbol(descriptor.Info.Version.String()))
}
```

Agree with your sentiment. Lint would have caught this. 😄

File: internal/logging/logging.go (hunk 2 above)
Comment (sclevine) [reply internal/logging/logging.go]: We should fix this as well -- definitely better to export (or return through an interface).

**Comment by dgageot** [comment]
@jromero I've updated the PR to make sure `golangci-lint` is installed by the Makefile. However, it pulls so many dependencies that I chose not to pollute the main `go.mod`. You tell me what you think

**Pull request merged**